From 516e6ab5fd01e6d1db52aee11424f67258d5cb7d Mon Sep 17 00:00:00 2001
From: Naveen M K <naveen521kk@gmail.com>
Date: Thu, 11 Jul 2024 13:58:20 +0530
Subject: [PATCH 078/N] configure: make `incompatible-pointer-types` a
 warning

gcc 14 changed `incompatible-pointer-types` from warning to error
for now, make it as warning to avoid build failure
---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 782395e..e1d428d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2657,6 +2657,14 @@ fi
 AS_VAR_IF([ac_cv_gcc_compat], [yes], [
     CFLAGS_NODIST="$CFLAGS_NODIST -std=c11"
 
+    # GCC 14 and later error out with incompatible-pointer-types
+    # this only occues on mingw
+    case $host in
+    *-*-mingw*)
+        CFLAGS_NODIST="$CFLAGS_NODIST -Wno-error=incompatible-pointer-types"
+      ;;
+    esac
+
     PY_CHECK_CC_WARNING([enable], [extra], [if we can add -Wextra])
     AS_VAR_IF([ac_cv_enable_extra_warning], [yes],
               [CFLAGS_NODIST="$CFLAGS_NODIST -Wextra"])
